processing/src/main/java/io/druid/segment/FloatDimensionSelector.java

```diff
@@ -0,0 +1,130 @@
+package io.druid.segment;
+
+import com.google.common.collect.Maps;
+import io.druid.query.extraction.ExtractionFn;
+import io.druid.segment.column.ColumnCapabilities;
+import io.druid.segment.data.IndexedFloats;
+import io.druid.segment.data.IndexedInts;
+import io.druid.segment.data.IndexedFloats;
+import io.druid.segment.data.IndexedLongs;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class FloatDimensionSelector implements DimensionSelector
+{
+  private final FloatColumnSelector selector;
+  private final ExtractionFn extractionFn;
+  private final ColumnCapabilities capabilities;
+  private final Map<Comparable, String> remappedValues = Maps.newHashMap();
+
+  public FloatDimensionSelector(Object selector, ExtractionFn extractionFn, ColumnCapabilities capabilities)
+  {
+    this.selector = (FloatColumnSelector) selector;
+    this.extractionFn = extractionFn;
+    this.capabilities = capabilities;
+  }
+
+  @Override
+  public IndexedInts getRow()
+  {
+    throw new UnsupportedOperationException("float column does not support getRow");
+  }
+
+  @Override
+  public int getValueCardinality()
+  {
+    return Integer.MAX_VALUE;
+  }
+
+  @Override
+  public int lookupId(String name)
+  {
+    throw new UnsupportedOperationException("float column does not support lookupId");
+  }
+
+  @Override
+  public String lookupName(int id)
+  {
+    throw new UnsupportedOperationException("float column does not support lookupName");
+  }
+
+  @Override
+  public IndexedFloats getFloatRow()
+  {
+    final float val = selector.get();
+    return new IndexedFloats()
+    {
+      @Override
+      public int size()
+      {
+        // only single-value numerics are supported
+        return 1;
```

Comment:
@fjy Added a size() > 1 check to LongDimensionIndexer and FloatDimensionIndexer, they'll throw an exception now if a multi-value row is passed to them